python/feathub/processors/flink/table_builder/flink_table_builder.py

```diff
@@ -578,14 +578,13 @@ def _evaluate_expression_transform(
         source_table: NativeFlinkTable,
         transform: ExpressionTransform,
         result_field_name: str,
-        result_type: DType,
+        result_type: Optional[DType],
     ) -> NativeFlinkTable:
-        result_type = to_flink_type(result_type)
-        return source_table.add_or_replace_columns(
-            native_flink_expr.call_sql(to_flink_sql_expr(transform.expr))
-            .cast(result_type)
-            .alias(result_field_name)
-        )
+        res_col = native_flink_expr.call_sql(to_flink_sql_expr(transform.expr))
+        if result_type is not None:
+            result_type = to_flink_type(result_type)
```

Comment:
With the implementation in this PR, we will leave the derivation of `Feature#dtype` to the underlying processor (e.g. Flink, Spark, Pandas). 

It means that FeatHub won't be able to tell what is the type of features emitted by its FeatureView. However, since FeatHub provides its own SDK and DSl for users to define schema, it should be able to derive and store the expected type of a FeatureView, just like what Spark/Flink does.

Also, would there be inconsistency between the dtypes derived by different processors?

Would it be better to derive FeatHub's Feature#dtype when we resolve a FeatureView and still ask the underlying processor to cast the feature to this dtype?